install/playbooks/roles/dovecot/tasks/main.yml

```diff
@@ -31,22 +31,31 @@
 ################################################################################
 # At this point, the certificates should have been created already #############
 # in order to have SSL and TLS encryption activated.                           #
-- name: Allow slapd and nslcd daemons to access the certificate directories
-  tags: dovecot,ssl
-  notify: Restart dovecot
-  acl:
-    path: '{{ path }}'
-    entity: dovecot
-    etype: user
-    permissions: rx
-    state: present
-    recursive: yes
-    default: yes
-  with_items:
-    - /etc/letsencrypt/archive
-    - /etc/letsencrypt/live
-  loop_control:
-    loop_var: path
+- name: Set the variables for the certificate
+  set_fact:
+    cert_dir: 'imap.{{ network.domain }}'
+    entity_group: dovecot
+
+- name: Set the default permissions for the imap certificate folders
+  include_tasks: '{{ playbook_dir }}/../../common/roles/cert-perms/tasks/main.yml'
+
+- name: Set the variables for the certificate
+  set_fact:
+    cert_dir: 'ldap.{{ network.domain }}'
+    entity_group: dovecot
+
+- name: Set the default permissions for the ldap certificate folders
+  include_tasks: '{{ playbook_dir }}/../../common/roles/cert-perms/tasks/main.yml'
+
+- name: Set the variables for the certificate
+  when: mail.pop3
+  set_fact:
+    cert_dir: 'pop3.{{ network.domain }}'
+    entity_group: dovecot
+
+- name: Set the default permissions for the pop3 certificate folders
+  when: mail.pop3
+  include_tasks: '{{ playbook_dir }}/../../common/roles/cert-perms/tasks/main.yml'
```

Comment:
Ha, ok, in the dovecot main tasks, the permissions will be set, assuming that the certificates are already generated by certbot. So, the directories are already created with the permissions set by certbot.

The certificates for dovecot are generated in separate playbooks and there is no dependency relationship between those playbooks and roles. Maybe this could be refactored in the future. (Integrated in the certificates role? Just looking at it, I am not sure of the implications)

But it seems quite some changes… If it works like that, it looks like better version of the current rights/ACL setup, so that's good enough for me for now.

Reply:
I thought about adding the permissions in the certificate roles, this would be indeed the cleanest way, unfortunately, the user that need access to the certificates is not existing yet!

However, I agree some refactoring is needed.

Thanks for the feedback